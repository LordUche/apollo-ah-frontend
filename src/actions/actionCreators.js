/* eslint-disable import/prefer-default-export */
import {
  PASSWORD_RESET_REQUEST_LOADING,
  PASSWORD_RESET_REQUEST_FAILURE,
  PASSWORD_RESET_REQUEST_SUCCESS,
} from './actionTypes';

/**
 * request loading
 *
 * @export
 * @param boolean isLoading - indicate if there is an ongoing request
 * @param boolean responseData - data gotten from the request
 * @returns {object}
 */

export function requestLoading(isLoading, responseData) {
  return {
    type: PASSWORD_RESET_REQUEST_LOADING,
    isLoading,
    responseData,
  };
}

/**
 * request loading success
 *
 * @export
 * @param boolean isLoading - indicate if there is an ongoing request
 * @param string responseData - response message gotten from the request
 * @returns {object}
 */

export function requestLoadingSuccess(isLoading, responseData) {
  return {
    type: PASSWORD_RESET_REQUEST_SUCCESS,
    isLoading,
    responseData,
  };
}

/**
 * request loading failure
 *
 * @export
 * @param boolean isLoading - indicate if there is an ongoing request
 * @param string responseData - response message gotten from the request
 * @returns {object}
 */

export function requestLoadingFail(isLoading, responseData) {
  return {
    type: PASSWORD_RESET_REQUEST_FAILURE,
    isLoading,
    responseData,
  };
}