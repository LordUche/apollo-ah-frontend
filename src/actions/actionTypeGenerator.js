const typeGenerator = type => type

export default typeGenerator;
