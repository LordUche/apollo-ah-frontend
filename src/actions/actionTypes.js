import typeGenerator from './typeGenerator';

export const authenticationType = typeGenerator('AUTHENTICATION');
export const getArticleType = typeGenerator('GET_ARTICLE');
