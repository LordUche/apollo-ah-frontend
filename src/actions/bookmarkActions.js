import request from '../utils/request';
import typeGenerator from './typeGenerator';

// const API_URL = process.env.API_BASE_URL;

export const bookmarkArticleType = typeGenerator('BOOKMARK_ARTICLE');

export const bookmarkLoading = (type, status) => ({ type, status });
export const bookmarkArticleProcess = (type, data) => ({ type, data });

export const bookmarkArticleGenerators = ({ slug, token }) => async (dispatch) => {
  let data;
  dispatch(bookmarkLoading(bookmarkArticleType.loading, true));
  return request({
    method: 'post',
    payload: null,
    token,
    route: `articles/${slug}/bookmarks`
  }).then((response) => {
    const feedback = response.data;
    if (feedback.message === 'successfully bookmarked this article') {
      const bookmarkedData = feedback.data;
      data = bookmarkedData;
      return dispatch(bookmarkArticleProcess(bookmarkArticleType.success, data));
    }
    return dispatch(bookmarkArticleProcess(bookmarkArticleType.success, feedback.message));
  }).catch((error) => {
    let errorMessage = 'Please check your network connection';
    if (error.response) {
      const { message } = error.response.data;
      errorMessage = message;
    }
    dispatch(bookmarkArticleProcess(bookmarkArticleType.failure, errorMessage));
  });
};
