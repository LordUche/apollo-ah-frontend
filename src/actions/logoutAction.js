import typeGenerator from './typeGenerator';

export const logoutType = typeGenerator('LOG_OUT');

export const logoutSuccess = () => ({
  type: logoutType.success,
});

export const logoutUser = () => (dispatch) => {
  dispatch(logoutSuccess());
};
