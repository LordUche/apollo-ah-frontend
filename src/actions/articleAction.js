import { GET_ARTICLES_SUCCESS, GET_ARTICLES_FAILURE } from './actionTypes';
import http from '../services/http';

export const ActionResponse = (actionTypes, articles) => ({
  type: actionTypes,
  payload: articles
});

export const getArticles = () => async(dispatch) => {
  try {
    const response = await http.get('/articles?size=12');
    const { data: { data } } = response;
    if (response.status === 404) {
      dispatch(ActionResponse(GET_ARTICLES_FAILURE, 'Article not found'));
    }
    dispatch(ActionResponse(GET_ARTICLES_SUCCESS, data));
  } catch (err) {
    dispatch(err);
  }
};
