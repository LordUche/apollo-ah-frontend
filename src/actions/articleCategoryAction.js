import { GET_ARTICLES_CATEGORY_SUCCESS, GET_ARTICLES_CATEGORY_FAILURE } from './actionTypes';
import http from '../services/http';

export const ActionResponse = (actionType, category) => ({
  type: actionType,
  payload: category
});

export const getArticlesCategory = () => async(dispatch) => {
  try {
    const response = await http.get('/categories?size=6');
    const { data: { data } } = response;
    if (response.status === 404) {
      dispatch(ActionResponse(GET_ARTICLES_CATEGORY_FAILURE, 'Category not found'));
    }
    dispatch(ActionResponse(GET_ARTICLES_CATEGORY_SUCCESS, data));
  } catch (err) {
    dispatch(err);
  }
};
