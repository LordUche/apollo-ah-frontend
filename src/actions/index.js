import { ADD_DATA } from './actionTypes';

/**
 * Dummy action
 *
 * @export
 * @param {object} payload - Data payload
 * @return {object}
 */
export default function addData(payload) {
  return { type: ADD_DATA, payload };
}
export * from './articleAction';
export * from './articleCategoryAction';
