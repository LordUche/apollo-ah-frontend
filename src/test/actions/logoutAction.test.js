import configureStore from 'redux-mock-store';
import { logoutSuccess, logoutType } from '../../actions/logoutAction';

const mockStore = configureStore();

describe('logoutAction test suite', () => {
  it('should match the expected action type', () => {
    const expected = {
      type: logoutType.success,
    };
    expect(logoutSuccess()).toEqual(expected);
  });

  it('dispatches logout action', () => {
    const store = mockStore({});

    store.dispatch(logoutSuccess());
    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });
});
