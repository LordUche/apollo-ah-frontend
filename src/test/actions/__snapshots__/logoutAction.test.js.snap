// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logoutAction test suite dispatches logout action 1`] = `
Array [
  Object {
    "type": "LOG_OUT_SUCCESS",
  },
]
`;
